frontend/main-app/src/components/FormViewer/FormViewer.tsx

```diff
@@ -22,30 +30,46 @@ export const FormViewer = (): React.JSX.Element => {
 
     if ("error" in result) {
       setFormErrors(result.error.formErrors.fieldErrors);
-      setFormAlert("error");
+      setFormAlert("validationError");
     } else {
-      // To do: Pass form submission to backend
       setFormErrors({});
-      setFormAlert("success");
+      postSubmission(formSchema, cleanedFormData, setFormAlert);
     }
   };
 
-  return (
-    <main className={classes.card}>
-      <form onSubmit={handleSubmit}>
-        <Heading level={1} size="xlarge">
-          {formSchema.title}
-        </Heading>
-        <FormComponents components={formSchema.components} errors={formErrors} />
-        <div className={classes.buttonContainer}>
-          <Button type="submit" size={"large"} fullWidth={false}>
-            Submit form
-            <TasklistSendFillIcon />
-          </Button>
-        </div>
-      </form>
-      {formAlert == "success" && <Alert severity="success">{t("form_viewer.success")}</Alert>}
-      {formAlert == "error" && <Alert severity="danger">{t("form_viewer.error")}</Alert>}
-    </main>
-  );
+  if (formSchema) {
+    return (
+      <main className={classes.card}>
+        <form onSubmit={handleSubmit}>
+          <Heading level={1} size="xlarge">
+            {formSchema.title}
+          </Heading>
+          <FormComponents components={formSchema.components} errors={formErrors} />
+          <div className={classes.buttonContainer}>
+            <Button type="submit" size={"large"} fullWidth={false}>
+              Submit form
+              <TasklistSendFillIcon />
+            </Button>
+          </div>
+        </form>
+        {formAlert == "success" && <Alert severity="success">{t("form_viewer.success")}</Alert>}
+        {formAlert == "validationError" && (
+          <Alert severity="danger">{t("form_viewer.validationError")}</Alert>
+        )}
+        {formAlert == "serverError" && (
+          <Alert severity="danger">{t("form_viewer.serverError")}</Alert>
+        )}
+      </main>
+    );
+  }
+
+  if (!formSchema) {
+    return (
+      <main className={classes.card}>
+        <Heading spacing>Form not found</Heading>
+        {formId && <Paragraph>Could not find form with ID {formId}.</Paragraph>}
+        {!formId && <Paragraph>No form Id was provided in the URL.</Paragraph>}
```

Comment:
Maybe a similar approach as the suggested change for `formAlert`?

Reply:
✅ 
```
export const getFormIdError = (formId: string, t: TFunction) => {
  if (formId) {
    return (
      <Paragraph>
        {t("form_viewer.formId.notFound")}
        {formId}.
      </Paragraph>
    );
  }
  return <Paragraph>{t("form_viewer.formId.notProvided")}</Paragraph>;
};
```